multicluster/test/e2e/framework.go

```diff
@@ -101,106 +79,130 @@ func (data *TestData) createClients() error {
-	client := data.getClientOfCluster(clusterName)
-	return deleteService(client, namespace, name)
+func (data *MCTestData) deleteService(clusterName, namespace, name string) error {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.DeleteService(namespace, name); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (data *TestData) deleteTestNamespace(timeout time.Duration) error {
-	return data.deleteNamespaceInAllClusters(multiClusterTestNamespace, timeout)
+func (data *MCTestData) getService(clusterName, namespace, name string) (*corev1.Service, error) {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		return d.GetService(namespace, name)
+	}
+	return nil, fmt.Errorf("clusterName %s not found", clusterName)
 }
 
-func (data *TestData) deleteNamespace(clusterName string, namespace string, timeout time.Duration) error {
-	client := data.getClientOfCluster(clusterName)
-	return deleteNamespace(client, namespace, timeout)
+func (data *MCTestData) deleteTestNamespaces(timeout time.Duration) error {
+	var failedClusters []string
+	for cluster, d := range data.clusterTestDataMap {
+		if err := d.DeleteNamespace(multiClusterTestNamespace, timeout); err != nil {
+			failedClusters = append(failedClusters, cluster)
+		}
+	}
+	if len(failedClusters) > 0 {
+		return fmt.Errorf("failed to delete Namespace %s in clusters %v", multiClusterTestNamespace, failedClusters)
+	}
+	return nil
 }
 
-func (data *TestData) deleteNamespaceInAllClusters(namespace string, timeout time.Duration) error {
-	for _, client := range data.clients {
-		if err := deleteNamespace(client, namespace, timeout); err != nil {
+func (data *MCTestData) deleteNamespace(clusterName, namespace string, timeout time.Duration) error {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.DeleteNamespace(namespace, timeout); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (data *TestData) createPod(clusterName string, name string, namespace string, ctrName string, image string, command []string,
+func (data *MCTestData) createPod(clusterName, name, namespace, ctrName, image string, command []string,
 	args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(pod *corev1.Pod)) error {
-	client := data.getClientOfCluster(clusterName)
-	return createPod(client, name, namespace, ctrName, image, command, args, env, ports, hostNetwork, mutateFunc)
-}
-
-func (data *TestData) getService(clusterName string, namespace string, serviceName string) (*corev1.Service, error) {
-	client := data.getClientOfCluster(clusterName)
-	return getService(client, namespace, serviceName)
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.CreatePodOnNodeInNamespace(name, namespace, "", ctrName, image, command, args, env, ports, hostNetwork, mutateFunc); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
```
if d, ok := data.clusterTestDataMap[clusterName]; ok {
        return d.CreatePodOnNodeInNamespace(name, namespace, "", ctrName, image, command, args, env, ports, hostNetwork, mutateFunc)
}
return fmt.Errorf("clusterName %s not found", clusterName)
```
Otherwise creating a Pod in non-existing cluster would return non error.

Reply:
Done